src/common/plist.mligo

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: 2021 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+#if !COMMON_PLIST_H
+#define COMMON_PLIST_H
+
+#include "../types.mligo"
+
+[@inline]
+let plist_mem (key, plist_o : proposal_key * proposal_doubly_linked_list option): bool =
+  match plist_o with
+    | None -> false
+    | Some plist ->
+        if (key = plist.first)
+          then true
+          else Map.mem (key, false) plist.map
+
+
+[@inline]
+let plist_insert (key, plist_o : proposal_key * (proposal_doubly_linked_list option)): proposal_doubly_linked_list option =
+  match plist_o with
+    | None -> Some
+        { first = key
+        ; last = key
+        ; map = (Big_map.empty : ((proposal_key * plist_direction), proposal_key) big_map)
+        }
+    | Some plist -> Some
+          { plist with
+            last = key
+          ; map =
+              Map.add (plist.last, true) key // Add next link of the old last key
```

Comment:
I think it would be better to alias `true` and `false` as something similar to `next` and `previous`, and use it in code.

Reply:
Updated this.

Reply:
Have this been update everywher, for example https://github.com/tezos-commons/baseDAO/pull/318/files/1a946282438110ec0b534b350413a353d8db1eca#diff-41e7ba34a244b5cdc2749fe8e91f9dfb4aede990d497e9fe2933ff48a22f049eR86 ?

Reply:
Sorry. Was looking at the wrong commit, it seems.